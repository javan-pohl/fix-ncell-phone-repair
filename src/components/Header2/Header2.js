import * as React from 'react'
import { Link, useStaticQuery, graphql } from 'gatsby'
import PropTypes from 'prop-types'
import CallBar from '../CallBar/CallBar'
import Navigation2 from '../Navigation2/Navigation2'
import {
  header,
  headerContainer,
  highlight,
  branding,
  white,
} from './Header2.module.css'

function getData() {
  const data = useStaticQuery(graphql`
    query {
      site {
        siteMetadata {
          phone
          menuLinks {
            link
            testid
            text
          }
        }
      }
    }
  `)

  const {
    site: {
      siteMetadata: { phone, menuLinks },
    },
  } = data

  return { phone, menuLinks }
}

function Header2({ currentUrl }) {
  const { phone, menuLinks } = getData()
  return (
    <header className={header}>
      <div className={headerContainer}>
        <CallBar phone={phone} />
        <div className={branding} data-testid="bizName">
          <Link to="/">
            <span id="highlight" className={highlight}>
              FixNcell
              {' '}
            </span>
            <span className={white}>iPhone Repair</span>
          </Link>
        </div>
        <Navigation2 menuLinks={menuLinks} currentUrl={currentUrl} />
      </div>
    </header>
  )
}

Header2.propTypes = {
  currentUrl: PropTypes.string,
}
Header2.defaultProps = {
  currentUrl: '/',
}
export default Header2
