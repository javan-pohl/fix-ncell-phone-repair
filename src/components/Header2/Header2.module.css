.header {
  font: 15px/1.5 Arial, Helvetica, sans-serif;
  width: 100%;
  background: #0098ff;
  color: #ffffff;
  min-height: 50px;
  margin: 0 0 0 0;
}
.headerContainer {
  width: 80%;
  margin-left: auto;
  margin-right: auto;
  display: flex;
  display: row;
  justify-content: space-between;
  align-items: center;
}

header a {
  text-decoration: none;
  color: #ffffff;
}

header .mobiletel {
  display: none;
}

.highlight {
  color: #000000;
  font-weight: bold;
}
header a:hover {
  color: #cccccc;
}
.branding {
  font-size: 30px;
  font-weight: bold;
  padding-bottom: 0px;
  padding-top: 0px;
  bottom: 0px;
}

.branding a {
  margin-bottom: 0px;
  bottom: 0px;
}
@media (max-width: 1080px) {
  .headerContainer {
    display: inline;
    text-align: center;
  }

  .branding {
    margin: 0px;
  }
}
