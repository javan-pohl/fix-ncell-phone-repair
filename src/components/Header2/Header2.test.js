import 'jsdom-global/register'
import '@testing-library/jest-dom'
import React from 'react'
import * as Gatsby from 'gatsby'
import renderer from 'react-test-renderer'

import Header2 from './Header2'

const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery')
useStaticQuery.mockImplementation(() => ({
  site: {
    siteMetadata: {
      title: 'FixNcell Phone Repair',
      menuLinks: [
        {
          testid: 'homeLink',
          link: '/',
          text: 'Home',
        },
        {
          testid: 'aboutLink',
          link: '/about',
          text: 'About',
        },
        {
          testid: 'repairsLink',
          link: '/iphone-repair',
          text: 'Repairs',
        },
        {
          testid: 'contactLink',
          link: '/contact',
          text: 'Contact Us',
        },
      ],
    },
  },
}))

beforeEach(() => {
  jest.clearAllMocks()
})

describe('Header2', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Header2 />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
