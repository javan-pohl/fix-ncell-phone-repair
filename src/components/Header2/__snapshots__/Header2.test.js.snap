// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header2 renders correctly 1`] = `
<header
  className="header"
>
  <div
    className="headerContainer"
  >
    <div
      className="callBar"
      data-testid="callBar"
    >
      <a
        data-testid="callBarNum"
        href="tel:3034216499"
      >
        â˜Ž Call or Text
         
        (303)421-6499
      </a>
    </div>
    <div
      className="branding"
      data-testid="bizName"
    >
      <a
        href="/"
      >
        <span
          className="highlight"
          id="highlight"
        >
          FixNcell
           
        </span>
        <span
          className="white"
        >
          iPhone Repair
        </span>
      </a>
    </div>
    <div
      className="navigation"
    >
      <nav>
        <ul>
          <li>
            <a
              data-testid="homeLink"
              href="/"
              style={
                Object {
                  "color": "white",
                }
              }
            >
              Home
            </a>
          </li>
          <li>
            <a
              data-testid="aboutLink"
              href="/about"
              style={
                Object {
                  "color": "black",
                }
              }
            >
              About
            </a>
          </li>
          <li>
            <a
              data-testid="repairsLink"
              href="/iphone-repair"
              style={
                Object {
                  "color": "black",
                }
              }
            >
              Repairs
            </a>
          </li>
          <li>
            <a
              data-testid="contactLink"
              href="/contact"
              style={
                Object {
                  "color": "black",
                }
              }
            >
              Contact Us
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
